native/cocos/ui/edit-box/EditBox-ios.mm

```diff
@@ -363,76 +267,315 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    [confirmBtn setTitleColor:[UIColor systemBlueColor]
+                  forState:UIControlStateNormal];
+    [confirmBtn setTitleColor:[UIColor darkTextColor]
+                  forState:UIControlStateHighlighted]; // Hight light state triggered when the button is tapped.
+    UIBarButtonItem *confirm = [[UIBarButtonItem alloc]initWithCustomView:confirmBtn];
+    
+    [toolbar setItems:@[textViewItem, confirm] animated:YES];
+    UIBarButtonItem* textViewBarButtonItem = [self setInputWidthOf:toolbar];
+    ((UITextView*)[inputbox inputOnView]).inputAccessoryView = toolbar;
+    
+    [inputbox setInputOnToolbar:textViewBarButtonItem.customView];
+    
+}
+- (void) addInputAccessoryViewForTextField: (InputBoxPair*)inputbox
+                                      with: (const cc::EditBox::ShowInfo*)showInfo{
+    CGRect safeView = getSafeAreaRect();
+    UIToolbar* toolbar = [[UIToolbar alloc]
+                          initWithFrame:CGRectMake(0,
+                                                   0,
+                                                   safeView.size.width,
+                                                   TEXT_LINE_HEIGHT + ITEM_MARGIN_HEIGHT)];
+    [toolbar setBackgroundColor:[UIColor darkGrayColor]];
+    
+    UITextField* textField = [[UITextField alloc] init];
+    textField.borderStyle = UITextBorderStyleRoundedRect;
+    textField.textColor = [UIColor blackColor];
+    textField.backgroundColor = [UIColor whiteColor];
+    textField.text = [NSString stringWithUTF8String:showInfo->defaultValue.c_str()];
+    TextFieldDelegate* delegate = [[TextFieldDelegate alloc] initWithPairs:[inputbox inputOnView] and:textField];
+    textField.delegate = delegate;
+    [textField addTarget:delegate action:@selector(textFieldDidChange:) forControlEvents:UIControlEventEditingChanged];
+    UIBarButtonItem *textFieldItem = [[UIBarButtonItem alloc] initWithCustomView:textField];
+    
+    if (!btnHandler){
+        btnHandler = [[ButtonHandler alloc] init];
+    }
+    UIButton *confirmBtn = [[UIButton alloc]
+                            initWithFrame:CGRectMake(0,
+                                                     0,
+                                                     BUTTON_WIDTH,
+                                                     BUTTON_HEIGHT)];
+    [confirmBtn addTarget:btnHandler
+                   action:@selector(buttonTapped:)
+         forControlEvents:UIControlEventTouchUpInside];
+    [confirmBtn setTitle:[NSString stringWithUTF8String:showInfo->confirmType.c_str()]
+             forState:UIControlStateNormal];
+    [confirmBtn setTitleColor:[UIColor systemBlueColor]
+                  forState:UIControlStateNormal];
+    [confirmBtn setTitleColor:[UIColor darkTextColor]
+                  forState:UIControlStateHighlighted]; // Hight light state triggered when the button is tapped.
+    UIBarButtonItem *confirm = [[UIBarButtonItem alloc]initWithCustomView:confirmBtn];
```

Comment:
Is `confirmBtn` a strong reference or not? If it is a strong reference, then should release it here, or will cause memory leak. Should check other places too.

Reply:
Add release logic when leave the blocks.